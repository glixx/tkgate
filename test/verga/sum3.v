module top;
  parameter n = 2;

  reg [n-1:0] a, b;
  reg ci;
  reg [n-1:0] s;
  reg [n:0] c;
  integer i,j;

  always @(*) 
	begin
	  c[n:1] <= a & b | a & c[n-1:0] | b & c[n-1:0];
	  s <= a ^ b ^ c[n-1:0];
	end

  initial
    begin
	$display("Starting SUM test");
        for (i = 0;i < 32;i = i + 1)
	  begin
	    a[0] = (i & 8'h1) != 0;
	    a[1] = (i & 8'h2) != 0;
	    b[0] = (i & 8'h4) != 0;
	    b[1] = (i & 8'h8) != 0;
	    c[0] = (i & 8'h10) != 0;
	    #1 $display("sum: %3b = %2b + %2b + %2b       (%d=%d+%d+%d)",{c[2],s},a,b,c[0],{c[2],s},a,b,c[0]);
	  end
    end
endmodule
