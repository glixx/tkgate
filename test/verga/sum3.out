Starting SUM test
sum:   0 =  0 +  0 +  0       (0=0+0+0)
sum:   1 =  1 +  0 +  0       (1=1+0+0)
sum:  10 = 10 +  0 +  0       (2=2+0+0)
sum:  11 = 11 +  0 +  0       (3=3+0+0)
sum:   1 =  0 +  1 +  0       (1=0+1+0)
sum:  10 =  1 +  1 +  0       (2=1+1+0)
sum:  11 = 10 +  1 +  0       (3=2+1+0)
sum: 100 = 11 +  1 +  0       (4=3+1+0)
sum:  10 =  0 + 10 +  0       (2=0+2+0)
sum:  11 =  1 + 10 +  0       (3=1+2+0)
sum: 100 = 10 + 10 +  0       (4=2+2+0)
sum: 101 = 11 + 10 +  0       (5=3+2+0)
sum:  11 =  0 + 11 +  0       (3=0+3+0)
sum: 100 =  1 + 11 +  0       (4=1+3+0)
sum: 101 = 10 + 11 +  0       (5=2+3+0)
sum: 110 = 11 + 11 +  0       (6=3+3+0)
sum:   1 =  0 +  0 +  1       (1=0+0+1)
sum:  10 =  1 +  0 +  1       (2=1+0+1)
sum:  11 = 10 +  0 +  1       (3=2+0+1)
sum: 100 = 11 +  0 +  1       (4=3+0+1)
sum:  10 =  0 +  1 +  1       (2=0+1+1)
sum:  11 =  1 +  1 +  1       (3=1+1+1)
sum: 100 = 10 +  1 +  1       (4=2+1+1)
sum: 101 = 11 +  1 +  1       (5=3+1+1)
sum:  11 =  0 + 10 +  1       (3=0+2+1)
sum: 100 =  1 + 10 +  1       (4=1+2+1)
sum: 101 = 10 + 10 +  1       (5=2+2+1)
sum: 110 = 11 + 10 +  1       (6=3+2+1)
sum: 100 =  0 + 11 +  1       (4=0+3+1)
sum: 101 =  1 + 11 +  1       (5=1+3+1)
sum: 110 = 10 + 11 +  1       (6=2+3+1)
sum: 111 = 11 + 11 +  1       (7=3+3+1)
